src/db.rs

```diff
@@ -38,315 +39,338 @@ impl Default for Connection {
-    trnsx.exec_drop(mqstr, (&pid,))?;
-    trnsx.commit()?;
-    Ok(())
+    sqlx::query(query.as_str()).execute(&dbpool).await?;
+    Ok(uuid)
 }
 
-pub fn delete_connection(uuid: Uuid) -> Result<Uuid> {
-    let mut dbconn = get_dbconn()?;
+pub async fn insert_policy(policy: &policy::Policy) -> Result<u64> {
+    let dbpool = get_dbpool().await?;
 
-    let mqstr = "DELETE from conn_bundle WHERE id = ?";
+    //let allowed_digests_json = serde_json::to_string(&policy.allowed_digests)?;
+    //let allowed_policy_json = serde_json::to_string(&policy.allowed_policies)?;
+    //let allowed_build_ids_json = serde_json::to_string(&policy.allowed_build_ids)?;
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    trnsx.exec_drop(mqstr, (uuid.as_hyphenated().to_string(),))?;
-    trnsx.commit()?;
-    Ok(uuid)
+    let query = format!(
+        "INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids, create_date, valid) VALUES(\'{:?}\', \'{:?}\', {}, {}, \'{:?}\', NOW(), 1)",
+        policy.allowed_digests,
+        policy.allowed_policies,
+        policy.min_fw_api_major,
+        policy.min_fw_api_minor,
+        policy.allowed_build_ids
+    );
+
+    let last_insert_row = sqlx::query(query.as_str())
+        .execute(&dbpool)
+        .await?
+        .last_insert_id();
+
+    let last_insert_id = last_insert_row.unwrap();
+
+    Ok(last_insert_id as u64)
 }
 
-pub fn get_connection(uuid: Uuid) -> Result<Connection> {
-    let mut dbconn = get_dbconn()?;
+pub async fn get_policy(pid: u64) -> Result<policy::Policy> {
+    let dbpool = get_dbpool().await?;
 
-    let uuidstr = uuid.as_hyphenated().to_string();
-    let mqstr = "SELECT policy, fw_api_major, fw_api_minor, fw_build_id, launch_description, fw_digest FROM conn_bundle WHERE id = ?";
-
-    let conres = dbconn.exec_map(
-        mqstr,
-        (uuidstr,),
-        |(policy, fw_api_major, fw_api_minor, fw_build_id, launch_description, fw_digest)| {
-            Connection {
-                policy,
-                fw_api_major,
-                fw_api_minor,
-                fw_build_id,
-                launch_description,
-                fw_digest,
-            }
-        },
-    )?;
-
-    Ok(conres[0].clone())
+    let query = format!("SELECT allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids FROM policy WHERE id = {} AND valid = 1", pid);
+    let row_vec = sqlx::query(query.as_str()).fetch_one(&dbpool).await?;
+    Ok(policy::Policy {
+        allowed_digests: serde_json::from_str(&row_vec.try_get::<String, _>(0)?).unwrap(),
+        allowed_policies: serde_json::from_str(&row_vec.try_get::<String, _>(1)?).unwrap(),
+        min_fw_api_major: row_vec.try_get::<i32, _>(2)? as u32,
+        min_fw_api_minor: row_vec.try_get::<i32, _>(3)? as u32,
+        allowed_build_ids: serde_json::from_str(&row_vec.try_get::<String, _>(4)?).unwrap(),
+    })
```

Comment:
1. Following my comment above about removing `serde_json::to_string`, here you decode with serde_json. The encoding should use the same tool.
2. Check if we can replace `unwrap()` calls with `?` to avoid panics (we can add a test of a bad JSON string in the DB, like `"df}sj,dl]"`, and checking that this function returns Err and doesn't panic).

Reply:
Replaced unwrap's with ?.  I will add the test for bad JSON.